packages/block-library/src/search/index.php

```diff
@@ -81,17 +82,19 @@ function render_block_core_search( $attributes ) {
 				$button_internal_markup = wp_kses_post( $attributes['buttonText'] );
 			}
 		} else {
+			$aria_label             = sprintf( 'aria-label="%s"', esc_attr( wp_strip_all_tags( $attributes['label'] ) ) );
```

Comment:
@ryelle I am not sure using aria-label in this context should be done. Honestly, I think this whole file should be refactored to include `$attributes['button_aria_label']` that way users wouldn't be tempted to go out of their way to break accessibility. My problem with your approach is this.

```
$attributes['buttonText'] = '<span class="icon-class"></span>';
```

Now, the button is back to being inaccessible. Technically, it's not an icon button, but it still allows HTML right? Hints the need to strip tags for aria-label?

As I said, I would rather see an approach where accessibility related attributes are given and then we won't have to try to guess if users could make it inaccessible. Of course users can make it inaccessible, but the difference would be passing `$attributes['buttonText']` vs. `$attributes['button_aria_label']`.

Thoughts? Does this make sense?

Reply:
Sure, I'm good with that. I'll make a follow-up PR once this is in.

Reply:
Sounds good, though I'd suggest making an issue first - it'll need design input for the aria label settings UI.